### ISSUE TITLE
Error with socket.gethostbyname() on macOS

### ISSUE THREAD

**Issue by ZeyadFathallah** [ilabs-kdc/bluesky#48]
## Issue Summary
In [gltraffic.py](https://github.com/ilabs-kdc/bluesky/blob/master/bluesky/ui/qtgl/gltraffic.py#L302),  [gui.py](https://github.com/ilabs-kdc/bluesky/commit/9b71378ba3c5f59d8813cb1e05dd6e95065550cd#diff-7ebd11f71f929da4cc4f3a45c7d8049ecc48a0b205d53e5a3fc8c078b836ab84R36) and [tidgui.py](https://github.com/ilabs-kdc/bluesky/commit/f54c8c195fa971e82332797ec8ceb589a16c312a#diff-713d44c42685bd9b873f027e3471e702d1c4c3344fa3eb599e5c5b79c06d1e5eR61).

Commits 44f24c1 and  9b71378 added `socket.gethostbyname(socket.gethostname())` usage which upon running` BlueSky.py` gives the following error:

```
Traceback (most recent call last):
  File "/Users/zeyad/Downloads/bluesky 01_05_2023/bluesky/ui/qtgl/glhelpers.py", line 761, in paintGL
    obj.draw()
  File "/Users/zeyad/Downloads/bluesky 01_05_2023/bluesky/ui/qtgl/gltraffic.py", line 307, in draw
    IP = socket.gethostbyname(socket.gethostname())
socket.gaierror: [Errno 8] nodename nor servname provided, or not known

or 

Traceback (most recent call last):
  File "/Users/zeyad/Downloads/bluesky 01_05_2023/BlueSky.py", line 117, in <module>
    main()
  File "/Users/zeyad/Downloads/bluesky 01_05_2023/BlueSky.py", line 101, in main
    qtgl.start(mode)
  File "/Users/zeyad/Downloads/bluesky 01_05_2023/bluesky/ui/qtgl/gui.py", line 59, in start
    if socket.gethostbyname(socket.gethostname()) in ['192.168.0.6', '192.168.0.8']:
socket.gaierror: [Errno 8] nodename nor servname provided, or not known
```

The same function is used elsewhere in the code, might be worth changing it everywhere.

Using `socket.getaddrinfo(gethostname, None)` gives the same error.

## Possible Solution
Using the fully qualified domain name seems to work -->  `IP = socket.getfqdn()`
Work = gives no errors. I think it affects the code's functionality though.

## Environment

- Operating System: macOS 13.3.1 (a)
- Python 3.8 (venv using PyCharm)

**Comment by ZeyadFathallah** [comment]
@BvDillen

**Comment by ZeyadFathallah** [comment]
Tested distributed simulation using `socket.getfqdn()` (on Windows) with no issues.

**Comment by ZeyadFathallah** [comment]
Tested distributed simulation using `socket.getfqdn()` (on Windows) with no issues.

Closed with affa9ca.
